redis/src/test/scala/zio/redis/ApiSpec.scala

```diff
@@ -39,10 +44,170 @@ object ApiSpec extends BaseSpec {
+            value       <- uuid
+            expireAfter <- durationInMillis(10000L)
+            _           <- set(key, value, None, None, None)
+            exp         <- expire(key, expireAfter)
+            ttl1        <- ttl(key).either
+            persisted   <- persist(key)
+            ttl2        <- ttl(key).either
+          } yield assert(exp)(isTrue) && assert(ttl1)(isRight) && assert(persisted)(isTrue) && assert(ttl2)(isLeft)
+        },
+        testM("expireAt followed by get and ttl") {
+          for {
+            key      <- uuid
+            value    <- uuid
+            timeout  <- instantForNextMillis(5000L)
+            _        <- set(key, value, None, None, None)
+            exp      <- expireAt(key, timeout)
+            response <- get(key)
+            ttl      <- ttl(key)
+          } yield assert(exp)(isTrue) &&
+            assert(response)(isSome) &&
+            assert(ttl.toMillis)(isLessThan(timeout.toEpochMilli))
+        },
+        testM("rename existing key") {
+          for {
+            key     <- uuid
+            newKey  <- uuid
+            value   <- uuid
+            _       <- set(key, value, None, None, None)
+            renamed <- rename(key, newKey).either
+            v       <- get(newKey)
+          } yield assert(renamed)(isRight) && assert(v)(isSome(equalTo(value)))
+        },
+        testM("try to rename non-existing key") {
+          for {
+            key     <- uuid
+            newKey  <- uuid
+            renamed <- rename(key, newKey).either
+          } yield assert(renamed)(isLeft)
+        },
+        testM("rename key to newkey if newkey does not yet exist") {
+          for {
+            key     <- uuid
+            newKey  <- uuid
+            value   <- uuid
+            _       <- set(key, value, None, None, None)
+            renamed <- renameNx(key, newKey)
+            v       <- get(newKey)
+          } yield assert(renamed)(isTrue) && assert(v)(isSome(equalTo(value)))
+        },
+        testM("try to rename non-existing key with RENAMENX command") {
+          for {
+            key     <- uuid
+            newKey  <- uuid
+            renamed <- renameNx(key, newKey).either
+          } yield assert(renamed)(isLeft)
+        },
+        testM("unlink existing key") {
+          for {
+            key     <- uuid
+            value   <- uuid
+            _       <- set(key, value, None, None, None)
+            removed <- unlink(key, Nil)
+          } yield assert(removed)(equalTo(1L))
+        },
+        testM("touch two existing keys") {
+          for {
+            key1    <- uuid
+            value1  <- uuid
+            _       <- set(key1, value1, None, None, None)
+            key2    <- uuid
+            value2  <- uuid
+            _       <- set(key2, value2, None, None, None)
+            touched <- touch(key1, List(key2))
+          } yield assert(touched)(equalTo(2L))
+        },
+        testM("set followed by scan") {
```

Comment:
No need to include set in a test description, i.e. treat set as necessary evil :).